### PR TITLE
update token price api

### PR DESCRIPTION
**Pull request by mintdart**

### CODE REVIEW

File: src/queries/useGetPrice.tsx [hunk 1]
```diff
@@ -17,28 +17,86 @@ interface IPrice {
...
+	if (chain === 'gnosis') return 'xdai';
+	return chain;
+}
+
+async function getCoinsPrice({ chain: rawChain, fromToken, toToken }: IGetPriceProps) {
+	let gasTokenPrice, fromTokenPrice, toTokenPrice;
+
+	try {
+		const cgPrices = await Promise.allSettled([
+			fetch(`https://api.coingecko.com/api/v3/simple/price?ids=${rawChain}&vs_currencies=usd
```
Comment (0xngmi) [inline src/queries/useGetPrice.tsx:30]: does platform name always match with coin name of the gas token? havent checked myself

File: src/queries/useGetPrice.tsx [hunk 2]
```diff
@@ -17,28 +17,86 @@ interface IPrice {
...
+		if (!toTokenPrice) {
+			llamaApi.push(`${rawChain}:${toToken}`);
+		}
+
+		if (llamaApi.length > 0) {
+			const { coins } = await fetch(`https://coins.llama.fi/prices/current/${llamaApi.join(',')}`).then((r) =>
+				r.json()
+			);
+
+			gasTokenPrice = gasTokenPrice || coins[`${rawChain}:${ZERO_ADDRESS}`]?.price;
```
Comment (0xngmi) [inline src/queries/useGetPrice.tsx:64]: llama api blockchain names and coingecko's blockchain names are different

**Comment by 0xngmi** [comment]
+ do we automatically refetch new prices regularly atm?

**Comment by mintdart** [comment]
> * do we automatically refetch new prices regularly atm?

yes, every 20 seconds

File: src/queries/useGetPrice.tsx (hunk 1 above)
Comment (mintdart) [reply src/queries/useGetPrice.tsx:30]: hmm I'll look into it, but coingecko doesn't return price of gastoken(zero address) like this`https://api.coingecko.com/api/v3/simple/token_price/ethereum?contract_addresses=0x0000000000000000000000000000000000000000&vs_currencies=usd`

File: src/queries/useGetPrice.tsx (hunk 2 above)
Comment (mintdart) [reply src/queries/useGetPrice.tsx:64]: okay will update

**Review by 0xngmi — APPROVED** [review]

**Pull request merged**